Incorrect module resolution

```
➜ ls ..
import-file-url  llrt  module-resolve
➜ ls
bar.mjs  foo.mjs
➜ bat ./*.mjs
───────┬───────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: ./bar.mjs
───────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ export function bar() {
   2   │   return "bar";
   3   │ }
───────┴───────────────────────────────────────────────────────────────────────────────────────────────────────────
───────┬───────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: ./foo.mjs
───────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ import { bar } from "../bar.mjs";
   2   │ 
   3   │ console.log(bar());
───────┴───────────────────────────────────────────────────────────────────────────────────────────────────────────
➜ node foo.mjs 
node:internal/modules/esm/resolve:283
    throw new ERR_MODULE_NOT_FOUND(
          ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/nomyfan/develop/demos/llrt-issues/bar.mjs' imported from /Users/nomyfan/develop/demos/llrt-issues/module-resolve/foo.mjs
    at finalizeResolution (node:internal/modules/esm/resolve:283:11)
    at moduleResolve (node:internal/modules/esm/resolve:952:10)
    at defaultResolve (node:internal/modules/esm/resolve:1188:11)
    at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:542:12)
    at ModuleLoader.resolve (node:internal/modules/esm/loader:510:25)
    at ModuleLoader.getModuleJob (node:internal/modules/esm/loader:239:38)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:96:40)
    at link (node:internal/modules/esm/module_job:95:36) {
  code: 'ERR_MODULE_NOT_FOUND',
  url: 'file:///Users/nomyfan/develop/demos/llrt-issues/bar.mjs'
}

Node.js v20.18.3
➜ LLRT_LOG=trace ../llrt foo.mjs 
2025-03-23T02:29:09.860Z TRACE llrt: Started runtime 
2025-03-23T02:29:09.863Z TRACE llrt: Initialized VM in 3ms 
2025-03-23T02:29:09.863Z TRACE llrt_core::module_loader::resolver: Try resolve 'foo.mjs' from 'eval_script' 
2025-03-23T02:29:09.863Z TRACE llrt_core::module_loader::resolver: require_resolve(x, y):(foo.mjs, eval_script) 
2025-03-23T02:29:09.864Z TRACE llrt_core::module_loader::resolver: +- Resolved by `FILE`: foo.mjs
 
2025-03-23T02:29:09.864Z TRACE llrt_core::module_loader::loader: Loading module: /Users/nomyfan/develop/demos/llrt-issues/module-resolve/foo.mjs 
2025-03-23T02:29:09.864Z TRACE llrt_core::module_loader::resolver: Try resolve '../bar.mjs' from '/Users/nomyfan/develop/demos/llrt-issues/module-resolve/foo.mjs' 
2025-03-23T02:29:09.864Z TRACE llrt_core::module_loader::resolver: require_resolve(x, y):(../bar.mjs, /Users/nomyfan/develop/demos/llrt-issues/module-resolve/foo.mjs) 
2025-03-23T02:29:09.864Z TRACE llrt_core::module_loader::resolver: +- Resolved by `FILE`: bar.mjs
 
2025-03-23T02:29:09.864Z TRACE llrt_core::module_loader::loader: Loading module: /Users/nomyfan/develop/demos/llrt-issues/module-resolve/bar.mjs 
bar
```
`bar.mjs` exists in the same directory as `foo.mjs`, but I can load `bar.mjs` in `foo.mjs` using the specifier `../bar.mjs`.

--- a/fixtures/test903/bar.mjs
+++ b/fixtures/test903/bar.mjs
@@ -1,0 +1,3 @@
+export function bar() {
+  return "bar";
+}

--- a/fixtures/test903/foo.mjs
+++ b/fixtures/test903/foo.mjs
@@ -1,0 +1,2 @@
+import { bar } from "../bar.mjs";
+console.log(bar());

--- a/llrt_core/src/module_loader/resolver.rs
+++ b/llrt_core/src/module_loader/resolver.rs
@@ -141,6 +141,10 @@ pub fn require_resolve<'a>(
     let (_, ext_name) = name_extname(x);
     let is_supported_ext = is_supported_ext(ext_name);
 
+    let x_is_absolute = path::is_absolute(x);
+    let x_starts_with_current_dir = x.starts_with("./");
+    let x_starts_with_parent_dir = x.starts_with("../");
+
     if is_supported_ext && Path::new(x).is_file() {
         return resolved_by_file_exists(x.into());
     }
@@ -150,12 +154,9 @@ pub fn require_resolve<'a>(
         return resolved_by_bytecode_cache(x_normalized.into());
     }
 
-    if is_supported_ext && Path::new(&x_normalized).is_file() {
+    if !x_starts_with_parent_dir && is_supported_ext && Path::new(&x_normalized).is_file() {
         return resolved_by_file_exists(x_normalized.into());
     }
-
-    let x_is_absolute = path::is_absolute(x);
-    let x_starts_with_current_dir = x.starts_with("./");
 
     // 2. If X begins with '/'
     let y = if path::is_absolute(x) {
@@ -174,7 +175,7 @@ pub fn require_resolve<'a>(
     };
 
     // 3. If X begins with './' or '/' or '../'
-    if x_starts_with_current_dir || x_is_absolute || x.starts_with("../") {
+    if x_starts_with_current_dir || x_is_absolute || x_starts_with_parent_dir {
         let y_plus_x = if x_is_absolute {
             x.into()
         } else if x_starts_with_current_dir {

--- a/tests/unit/require.test.ts
+++ b/tests/unit/require.test.ts
@@ -153,6 +153,12 @@ it("require `@aws-lambda-powertools` module element", () => {
   );
 });
 
+it("regression testing for issue #903", () => {
+  expect(() => _require(`${CWD}/fixtures/test903/foo.mjs`)).toThrow(
+    /Error resolving module /
+  );
+});
+
 //create a test that spawns a subprocess and executes require.mjs from fixtures and captures stdout
 it("should handle blocking requires", (done) => {
   const proc = spawn(process.argv0, [`${CWD}/fixtures/require.mjs`]);

